conductor/lib/uploader.py

```diff
@@ -261,28 +325,44 @@ def do_work(self, job, thread_int):
             raise
 
     @common.DecRetry(retry_exceptions=api_client.CONNECTION_EXCEPTIONS, tries=5)
-    def do_upload(self, upload_url, filename, md5):
+    def do_upload(self, upload_url, filename, file_size, md5):
         '''
         Note that for GCS we don't rely on the make_request's own retry mechanism because
         we need to recreate the chunked_reader generator before retrying the request.
         Instead, we wrap this method in a retry decorator.
 
-        We cannot reuse make_request method for S3 because it adds auth headers that
-        S3 does not accept. S3 also requires additional signatures to support chunked data.
-        For now S3 uploads will read all data in memory resulting in a smaller supported size.
+        We cannot reuse make_request method for S3 because it adds auth and Transfer-Encoding headers that
+        S3 does not accept.
         '''
 
         if "amazonaws" in upload_url:
-            headers = {'Content-Type': 'application/octet-stream'}
+            headers = {
+                'Content-Type': 'application/octet-stream',
+                'Content-Length': str(file_size),
+            }
 
             with open(filename, 'rb') as fh:
-                # TODO: update make_request to be flexible with auth headers
-                # TODO: support chunked or streamed data
-                return self.api_client._make_request(verb="PUT",
-                                                     conductor_url=upload_url,
-                                                     headers=headers,
-                                                     params=None,
-                                                     data=fh)
+                # TODO: support chunked
+                response = self.api_client.make_prepared_request(
+                    verb="PUT",
+                    url=upload_url,
+                    headers=headers,
+                    params=None,
+                    data=fh,
+                    stream=True,
```

Comment:
Is there a reason we're setting `stream=True`?  I am not intimately familiar with its use, but I think it's used for streaming the content of the _response_ (rather than the content of the _request_).  So in our case, I don't believe it offers much utility. I'm a little worried that it might actually add micro overhead for each request. See [here](https://2.python-requests.org/en/v2.8.1/user/advanced/#body-content-workflow)

Reply:
There are still some benefits to using `stream=True`, since we are unable to use the chunk_reader with singlepart (len and Transfer-Encoding) or multipart (chunking based on partSize). It should help files being read without loading them completely into memory [stream uploads](https://2.python-requests.org/en/v2.8.1/user/advanced/#streaming-uploads).

Reply:
Can certainly default it to False on the api_client method.

Reply:
I'll set it as default to False on the api_client method and remove the kwarg.